examples/MultipleChains.md

```diff
@@ -0,0 +1,276 @@
+```
+
+```{code-cell} ipython3
+loc, scale = 10, 20
+observed = np.random.normal(loc, scale, size=1_000)
+
+
+def logprob_fn(loc, scale, observed=observed):
+    """Univariate Normal"""
+    logpdf = stats.norm.logpdf(observed, loc, scale)
+    return jnp.sum(logpdf)
+
+
+def logprob(x):
+    return logprob_fn(**x)
+
+
+def inference_loop(rng_key, kernel, initial_state, num_samples):
+
+    @jax.jit
+    def one_step(state, rng_key):
+        state, _ = kernel(rng_key, state)
+        return state, state
+
+    keys = jax.random.split(rng_key, num_samples)
+    _, states = jax.lax.scan(one_step, initial_state, keys)
+
+    return states
+```
+
+```{code-cell} ipython3
+inv_mass_matrix = np.array([0.5, 0.5])
+step_size = 1e-3
+
+nuts = blackjax.nuts(logprob, step_size, inv_mass_matrix)
+```
+
+```{code-cell} ipython3
+rng_key = jax.random.PRNGKey(0)
+num_chains = multiprocessing.cpu_count()
+```
+
+### Using `vmap`
+
+Here we apply `vmap` inside the `one_step` function, vectorizing the transition function,
+such that it can handle multiple states (and rng keys) at the same time.
+
+```{code-cell} ipython3
+def inference_loop_multiple_chains(
+    rng_key, kernel, initial_state, num_samples, num_chains
+):
+
+    @jax.jit
+    def one_step(states, rng_key):
+        keys = jax.random.split(rng_key, num_chains)
+        states, _ = jax.vmap(kernel)(keys, states)
+        return states, states
+
+    keys = jax.random.split(rng_key, num_samples)
+    _, states = jax.lax.scan(one_step, initial_state, keys)
+
+    return states
+```
+
+We now prepare the initial states (using `vmap` again to call `init` on a batch of initial positions)
+
+```{code-cell} ipython3
+initial_positions = {"loc": np.ones(num_chains), "scale": 2.0 * np.ones(num_chains)}
+initial_states = jax.vmap(nuts.init, in_axes=(0))(initial_positions)
+```
+
+And finally run the sampler
+
+```{code-cell} ipython3
+%%time
+
+states = inference_loop_multiple_chains(
+    rng_key, nuts.step, initial_states, 2_000, num_chains
+)
+_ = states.position["loc"].block_until_ready()
+```
+
+You can now access the samples from individual chains by simply indexing the returned arrays:
+
+```{code-cell} ipython3
+states.position["loc"].shape
+```
+
+E.g. to get the `loc` samples for the second chain:
+
+```{code-cell} ipython3
+samples = states.position["loc"][:, 1]
+samples
+```
+
+```{code-cell} ipython3
+samples.shape
+```
+
+```{code-cell} ipython3
+fig, ax = plt.subplots(figsize=(15, 6))
+
+ax.plot(states.position["loc"][:, 0], color="blue", alpha=0.25)
+ax.plot(states.position["loc"][:, 1], color="red", alpha=0.25)
+ax.set_xlabel("Samples")
+ax.set_ylabel("loc")
+ax.legend(["Chain 1", "Chain 2"])
+```
+
+### Using `pmap`
+
+In case of `pmap`, we can simply choose to apply the transformation directly to the original `inference_loop` function.
+
+```{code-cell} ipython3
+inference_loop_multiple_chains = jax.pmap(inference_loop, in_axes=(0, None, 0, None), static_broadcasted_argnums=(1, 3))
```

Comment:
What happens if you apply `jax.vmap` to `inference_loop`?

Reply:
Performance-wise it's exactly the same on my machine. 
IMO the code is then a bit cleaner (no need to write a new function).

Should I change the example and apply `vmap` to `inference_loop`? I didn't change the original example since I thought there might be some other reasons to do manually implement `inference_loop_multiple_chains`